sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -224,7 +224,11 @@ private object PostgresDialect extends JdbcDialect with SQLConfHelper {
         sqlException.getSQLState match {
           // https://www.postgresql.org/docs/14/errcodes-appendix.html
           case "42P07" => throw new IndexAlreadyExistsException(message, cause = Some(e))
-          case "42704" => throw new NoSuchIndexException(message, cause = Some(e))
+          case "42704" =>
+            // The message is: Failed to drop index indexName in tableName
+            val regex = "(?s)Failed to drop index (.*) in".r
```

Comment:
Seems we may be able to pick up the qualifiers as well ?

Reply:
Do you mean `tableName` so maybe do we want to show the error message like:

`Cannot find the index <indexName> on table <tableName>.`

?

Reply:
Yeah..

Reply:
Thanks! Just update the message